exchanges/kline/weighted_price_test.go

```diff
@@ -0,0 +1,310 @@
+	{Time: time.Date(2019, 10, 10, 9, 49, 0, 0, time.UTC), Open: 247.49, High: 247.65, Low: 247.49, Close: 247.65, Volume: 71995},
+	{Time: time.Date(2019, 10, 10, 9, 50, 0, 0, time.UTC), Open: 247.685, High: 247.801, Low: 247.65, Close: 247.69, Volume: 46038},
+	{Time: time.Date(2019, 10, 10, 9, 51, 0, 0, time.UTC), Open: 247.95, High: 248.74, Low: 247.95, Close: 248.74, Volume: 103773},
+	{Time: time.Date(2019, 10, 10, 9, 52, 0, 0, time.UTC), Open: 248.56, High: 248.56, Low: 247.95, Close: 247.95, Volume: 73810},
+	{Time: time.Date(2019, 10, 10, 9, 53, 0, 0, time.UTC), Open: 247.93, High: 247.93, Low: 247.6614, Close: 247.6614, Volume: 29784},
+	{Time: time.Date(2019, 10, 10, 9, 54, 0, 0, time.UTC), Open: 247.74, High: 247.76, Low: 247.65, Close: 247.76, Volume: 37138},
+	{Time: time.Date(2019, 10, 10, 9, 55, 0, 0, time.UTC), Open: 247.93, High: 248.03, Low: 247.93, Close: 248.03, Volume: 53166},
+	{Time: time.Date(2019, 10, 10, 9, 56, 0, 0, time.UTC), Open: 247.91, High: 248.44, Low: 247.91, Close: 248.44, Volume: 40789},
+	{Time: time.Date(2019, 10, 10, 9, 57, 0, 0, time.UTC), Open: 248.52, High: 248.52, Low: 248.3154, Close: 248.3154, Volume: 51988},
+	{Time: time.Date(2019, 10, 10, 9, 58, 0, 0, time.UTC), Open: 248.4409, High: 248.62, Low: 248.4409, Close: 248.62, Volume: 53405},
+	{Time: time.Date(2019, 10, 10, 9, 59, 0, 0, time.UTC), Open: 248.9199, High: 248.9199, Low: 248.9199, Close: 248.9199, Volume: 85348},
+	{Time: time.Date(2019, 10, 10, 10, 0, 0, 0, time.UTC), Open: 248.91, High: 249.08, Low: 248.42, Close: 248.72, Volume: 58270},
+}
+
+func TestGetVWAPs(t *testing.T) {
+	t.Parallel()
+	candles := Item{}
+	if _, err := candles.GetVWAPs(); !errors.Is(err, errNoData) {
+		t.Fatal(err)
+	}
+
+	candles.Candles = vwapdataset
+	vwap, err := candles.GetVWAPs()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if vwap[0] != 245.05046666666664 &&
+		vwap[1] != 245.00156932123465 &&
+		vwap[2] != 245.07320400593073 &&
+		vwap[3] != 245.19714781780763 &&
+		vwap[4] != 245.248374356565 &&
+		vwap[5] != 245.35797872352975 &&
+		vwap[6] != 245.45540807301208 &&
+		vwap[7] != 245.57298124760712 &&
+		vwap[8] != 245.61797546720302 &&
+		vwap[9] != 245.6901232761351 &&
+		vwap[10] != 245.7435986712912 &&
+		vwap[11] != 245.76128302894574 &&
+		vwap[12] != 245.771994363731 &&
+		vwap[13] != 245.7768929849006 &&
+		vwap[14] != 245.80115004533573 &&
+		vwap[15] != 245.82471633454026 &&
+		vwap[16] != 245.90964645148168 &&
+		vwap[17] != 246.0356579876492 &&
+		vwap[18] != 246.20233204964117 &&
+		vwap[19] != 246.29892677543359 &&
+		vwap[20] != 246.57315726207088 &&
+		vwap[21] != 246.70305234595537 &&
+		vwap[22] != 246.73669536160304 &&
+		vwap[23] != 246.7746731036053 &&
+		vwap[24] != 246.83849361010806 &&
+		vwap[25] != 246.89338504378165 &&
+		vwap[26] != 246.96313273581723 &&
+		vwap[27] != 247.03640100225914 &&
+		vwap[28] != 247.16505290840146 &&
+		vwap[29] != 247.23522648930867 {
+		t.Fatal("unexpected values")
+	}
```

Comment:
This would be much nicer with the normal `Received '%v' Expected '%v'` here. On a test fail, you have no idea which value is unexpected and what it is, making debugging unfun